sql: plumb vector search into execution logic

DrewKimball opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/138797)
<!-- Reviewable:end -->

DrewKimball committed:
opt: rename VectorPartitionSearch

This commit renames the `VectorPartitionSearch` operator to
`VectorMutationSearch`. This reflects the fact that the operator is
used to prepare the information needed for index updates.

Epic: None

Release note: None

DrewKimball committed:
opt: modify VectorMutationSearch to produce quantized vectors

This commit changes `VectorMutationSearch` so that it produces quantized
and encoded vectors, rather than partition centroids. This shifts the
responsibility of quantizing vectors to the `VectorMutationSearch`
operator rather than the insert/update code paths.

Epic: CRDB-42943

Release note: None

DrewKimball committed:
sql: plumb vector search into execution logic

DrewKimball commented:
Closing in favor of #140770 as the source of truth for unmerged work.
